dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -88,10 +94,12 @@ class NVDECCache {
 
       auto caps = CUVIDDECODECAPS{};
       caps.eCodecType = codec_type;
-      caps.eChromaFormat = cudaVideoChromaFormat_420;
-      caps.nBitDepthMinus8 = 0;
+      caps.eChromaFormat = chroma_format;
+      caps.nBitDepthMinus8 = bit_depth_luma_minus8;
       CUDA_CALL(cuvidGetDecoderCaps(&caps));
 
+      DALI_ENFORCE(caps.bIsSupported, "Codec not supported on this GPU");
+
       DALI_ENFORCE(width >= caps.nMinWidth  && height >= caps.nMinHeight,
```

Comment:
Would it make sense to expand those errors here so they list what we encountered in the video and what the GPU offers?

Reply:
Done